### PR TITLE
(BKR-624) Added a setter for line_prefix

### PR DESCRIPTION
**Pull request by dylanratcliffe**
Added a setter for `line_prefix` that calculates the length as it is set. Also modified the `step_in` and `step_out` methods to use this new setter

### CODE REVIEW

File: lib/beaker/logger.rb [hunk 1]
```diff
@@ -89,6 +89,12 @@ def initialize(*args)
       dests.each {|dest| add_destination(dest)}
     end
 
+    # Setter for line_prefix that also calculates line_prefix_length
+    # @param [String] prefix The line prefix
+    def line_prefix=(prefix)
+      @line_prefix = prefix
+      @line_prefix_length = prefix.length
```
Comment (kevpl) [inline lib/beaker/logger.rb]: if we're going to give up `@line_prefix_length`, we should do so entirely.  Why is it being kept/updated here?

**Comment by kevpl** [comment]
@dylanratcliffe please add a spec test, so that we can verify that we don't break this easily in the future.

File: lib/beaker/logger.rb (hunk 1 above)
Comment (dylanratcliffe) [inline lib/beaker/logger.rb]: @kevpl I was doing it for backwards compatibility bit it seems nothing is using it so i'll kill it

**Comment by dylanratcliffe** [comment]
@kevpl Okay all done

**Pull request merged**